### PR TITLE
ssh/tailssh: detect user shell correctly on darwin

### PR DESCRIPTION
**Pull request by maisem**
Updates #6213

### CODE REVIEW

File: ssh/tailssh/incubator.go [hunk 1]
```diff
@@ -572,15 +572,23 @@ func (ss *sshSession) startWithStdPipes() (err error) {
...
-		out, _ := exec.Command("getent", "passwd", uid).Output()
+		out, _ := exec.Command("getent", "passwd", u.Uid).Output()
 		// out is "root:x:0:0:root:/root:/bin/bash"
 		f := strings.SplitN(string(out), ":", 10)
 		if len(f) > 6 {
 			return strings.TrimSpace(f[6]) // shell
 		}
+	case "darwin":
+		// Note: /Users/username is key, and not the same as u.HomeDir.
+		out, _ := exec.Command("dscl", ".", "-read", filepath.Join("/Users", u.Username), "UserShell").Output()
```
Comment (andrew-d) [inline ssh/tailssh/incubator.go:586]: Does this work in the macOS sandbox?

**Review by andrew-d — APPROVED** [review]

File: ssh/tailssh/incubator.go (hunk 1 above)
Comment (maisem) [reply ssh/tailssh/incubator.go:586]: No, but we disallow that elsewhere

Comment (andrew-d) [reply ssh/tailssh/incubator.go:586]: SGTM; thanks!

**Comment by philiptaron** [comment]
This patch contained what I needed to solve [a thorny remote-building-using-Nix issue](https://github.com/NixOS/nix/issues/7508#issuecomment-2010693267). Thanks!

**Pull request merged**